### ISSUE TITLE
Replace vapory with an alternative solution.

### ISSUE THREAD

**Issue by yakutovicha** [aiidalab/aiidalab-widgets-base#200]
Despite vapory is a nice tool, it looks pretty abandoned. The issues are not addressed, the last release is in 2014.
The [request](https://github.com/Zulko/vapory/issues/57) for making a new release remains unanswered since October 2020.
Therefore, we should consider an alternative solution.

@csadorf pointed out a nice library called [Fresnel](https://fresnel.readthedocs.io/en/stable/). It looks very well maintained and, importantly, has all the features we are currently using for generating images with povray.

**Comment by dou-du** [comment]
> @dou-du would you be able to address that issue?

Sure, I will take care of it.

**Comment by dou-du** [comment]
Since now the POVRAY rendering is working fine, I think we can keep the rendering engine. It there is any problem in the future, we can replace it with other tools like Fresnel, beautiful-atoms etc.

**Comment by dou-du** [comment]
> I still see one issue with POVRAY - it is expensive.
> 
> It took 40 seconds to generate an image with 244 atoms on the AiiDAlab machine. Could we add an option of downloading the POVRAY script so that the user can generate the image locally?
> 
> Ideally, the tool should inform the user when things look too heavy and suggest running rendering somewhere else.
> 
> @dou-du what do you think?

We can set a limitation for the number of atoms to render. If the structure is large than (for example 100 atoms), it will disable the render button and given error information.

**Comment by yakutovicha** [comment]
> We can set a limitation for the number of atoms to render. If the structure is large than (for example 100 atoms), it will disable the render button and given error information.

Ok, but can we let the user download the povray script so they do not end up in a stalemate?

**Comment by dou-du** [comment]
> > We can set a limitation for the number of atoms to render. If the structure is large than (for example 100 atoms), it will disable the render button and given error information.
> 
> Ok, but can we let the user download the povray script so they do not end up in a stalemate?

Yes, it is possible. Let's to see how other people think in today's AiiDAlab meeting.

**Comment by unkcpz** [comment]
Hi @dou-du, can you update me on the progress?

> Let's to see how other people think in today's AiiDAlab meeting.

BTW, what the result we reached in the meeting? I don't remember.

**Comment by dou-du** [comment]
> Hi @dou-du, can you update me on the progress?
> 
> > Let's to see how other people think in today's AiiDAlab meeting.
> 
> BTW, what is the result we reached in the meeting? I don't remember.

I cannot remember either. However, I think it is not necessary to download the POVRAY scripts, since there will be not many people to use it.

**Comment by unkcpz** [comment]
I'll let @yakutovicha comment on this and see if we can close this issue since the original one is resolved.

**Comment by yakutovicha** [comment]
> I'll let @yakutovicha comment on this and see if we can close this issue since the original one is resolved.

I don't remember either, but I would still implement that. For large systems, the povray rendering is a challenge.
